go/tasks/plugins/k8s/sidecar/sidecar.go

```diff
@@ -73,45 +73,42 @@ type sidecarJob struct {
 	PrimaryContainerName string
 }
 
-func unmarshalSidecarCustom(structObj *structpb.Struct, sidecarJob *sidecarJob) error {
-	if structObj == nil {
-		return fmt.Errorf("nil Struct Object passed")
-	}
-
-	jsonObj, err := json.Marshal(structObj)
-	if err != nil {
-		return err
-	}
-
-	if err = json.Unmarshal(jsonObj, sidecarJob); err != nil {
-		return err
-	}
-
-	return nil
-}
-
 func (sidecarResourceHandler) BuildResource(ctx context.Context, taskCtx pluginsCore.TaskExecutionContext) (client.Object, error) {
-	sidecarJob := sidecarJob{}
+	var podSpec k8sv1.PodSpec
+	var primaryContainerName string
+
 	task, err := taskCtx.TaskReader().Read(ctx)
 	if err != nil {
 		return nil, errors.Errorf(errors.BadTaskSpecification,
 			"TaskSpecification cannot be read, Err: [%v]", err.Error())
 	}
-	err = unmarshalSidecarCustom(task.GetCustom(), &sidecarJob)
-	if err != nil {
-		return nil, errors.Errorf(errors.BadTaskSpecification,
-			"invalid TaskSpecification [%v], Err: [%v]", task.GetCustom(), err.Error())
+	if task.TaskTypeVersion == 0 {
+		sidecarJob := sidecarJob{}
+		err := utils.UnmarshalStructToObj(task.GetCustom(), &sidecarJob)
+		if err != nil {
+			return nil, errors.Errorf(errors.BadTaskSpecification,
+				"invalid TaskSpecification [%v], Err: [%v]", task.GetCustom(), err.Error())
+		}
+		podSpec = *sidecarJob.PodSpec
+		primaryContainerName = sidecarJob.PrimaryContainerName
+	} else {
+		err := utils.UnmarshalStructToObj(task.GetCustom(), &podSpec)
+		if err != nil {
+			return nil, errors.Errorf(errors.BadTaskSpecification,
+				"Unable to unmarshal task custom [%v], Err: [%v]", task.GetCustom(), err.Error())
+		}
+		primaryContainerName = task.GetConfig()[primaryContainerKey]
```

Comment:
what if this is missing?, check for that
```suggestion
		primaryContainerName, ok = task.GetConfig()[primaryContainerKey]
```

Reply:
done